### DISCUSSION TITLE
How to disable telemetry such as NetSecurityTelemetry

### DISCUSSION THREAD

**Discussion by HighPerfDotNet** [dotnet/runtime#96842] [General] [answered]
Whilst debugging code I've come across with .NET's source throwing exception and using NetSecurityTelemetry.Log.IsEnabled() to choose code path which would log stuff via that NetSecurityTelemetry. This can't be the only part of the library that uses this logging.

I've set DOTNET_CLI_TELEMETRY_OPTOUT=1 and restarted VS 2022 but that had no effect on this, as it seeminly affects only CLI, 

How can this and all other unwanted logging get disabled, ideally programmatically?

**Comment by jkotas** [comment]
NetSecurityTelemetry is based on [EventSource](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource). EventSource tracing is disabled by default. It gets enabled when a listener registers for given event. An example of such listener is [dotnet-trace](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace) tool.

**Comment by HighPerfDotNet** [comment]
Does it get automatically enabled in Debug mode running under VS 2022?

**Reply by jkotas** [reply, marked as answer]
Yes. the managed Visual Studio Debugger enables number of event sources to implement various debugger features.
